chain/yoda/gas.go

```diff
@@ -0,0 +1,108 @@
+package yoda
+
+import (
+	"github.com/bandprotocol/bandchain/chain/x/oracle/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Constant used to estimate gas price of reports transaction.
+const (
+	// cosmos
+	baseFixedGas        = uint64(37764)
+	baseTransactionSize = uint64(200)
+	txCostPerByte       = uint64(5) // Using DefaultTxSizeCostPerByte of BandChain
+
+	readingBaseCost = uint64(1000)
+	writingBaseCost = uint64(2000)
+
+	readingCostPerByte = uint64(3)
+	writingCostPerByte = uint64(30)
+
+	// band
+	baseReportCost     = uint64(4006)
+	payingFeeCost      = uint64(15200)
+	countingReportCost = uint64(1479)
+	addingPendingCost  = uint64(3400)
+
+	baseRequestSize = uint64(32)
+	addressSize     = uint64(20)
+
+	baseRawRequestSize = uint64(16)
+)
+
+func estimateTxSize(c Context, msgs []sdk.Msg) uint64 {
+	// base tx + reports
+	size := baseTransactionSize
+
+	for _, msg := range msgs {
+		msg, ok := msg.(types.MsgReportData)
+		if !ok {
+			panic("Don't support non-report data message")
+		}
+
+		ser := cdc.MustMarshalBinaryBare(msg)
+		size += uint64(len(ser))
+	}
+
+	return size
+}
+
+func estimateStoringReportCost(c Context, msg sdk.Msg) uint64 {
+	cost := writingBaseCost
+	cost += uint64(len(cdc.MustMarshalBinaryBare(msg))) * writingCostPerByte
```

Comment:
This is the estimated cost of report in storage not a real one right?

Reply:
This is real value that would be stored.